openmetadata-service/src/main/java/org/openmetadata/service/apps/bundles/changeEvent/AbstractEventConsumer.java

```diff
@@ -256,18 +256,34 @@ public List<ChangeEvent> pollEvents(long offset, long batchSize) {
   public void execute(JobExecutionContext jobExecutionContext) {
     // Must Have , Before Execute the Init, Quartz Requires a Non-Arg Constructor
     this.init(jobExecutionContext);
-    // Poll Events from Change Event Table
-    List<ChangeEvent> batch = pollEvents(offset, eventSubscription.getBatchSize());
-    int batchSize = batch.size();
-    Map<ChangeEvent, Set<UUID>> eventsWithReceivers = createEventsWithReceivers(batch);
+    long batchSize = 0;
+    Map<ChangeEvent, Set<UUID>> eventsWithReceivers = new HashMap<>();
     try {
+      // Poll Events from Change Event Table
+      List<ChangeEvent> batch = pollEvents(offset, eventSubscription.getBatchSize());
+      batchSize = batch.size();
+      eventsWithReceivers.putAll(createEventsWithReceivers(batch));
       // Publish Events
       if (!eventsWithReceivers.isEmpty()) {
         alertMetrics.withTotalEvents(alertMetrics.getTotalEvents() + eventsWithReceivers.size());
         publishEvents(eventsWithReceivers);
       }
     } catch (Exception e) {
-      LOG.error("Error in executing the Job : {} ", e.getMessage());
+      long currentLatest = Entity.getCollectionDAO().changeEventDAO().getLatestOffset();
```

Comment:
do we need to make a call to db here?

Reply:
isnt' offset is the current offset and all we need to increment the batch?

Reply:
Ex: - Change event has 50 entries table
current offset is let's say 20 , batch size 10
reading from 20 to  with batch size 10 if the fails , then i update the offset to 20 + 10 because 50 (total or latest offset) - current > batch size so we next batch of events
but let's say current is 45 and batch is 10 
reading from 45 there is a failure in this case 50-45 < batch size so the current offset needs to be updated to offset + remanining items(5)

Reply:
and at any failure we need the latest offset of change_event

Reply:
we do have offset of the message no? 
if we record what is the offset its failing then we should be able to move currentOffset = failedOffset+1.
Trrying to see if we can avoid polling db

Reply:
it's in batch , we are reading in batches which are passed , so it is difficult to add per message at that layer

Reply:
can we throw a custom exception to catch, exception should contain the offset it failed on?

Reply:
done , removed the db call as well